src/rules.spec.ts

```diff
@@ -0,0 +1,323 @@
+import { encodeBase64, getMD5Hash } from './obfuscation';
+import { ObfuscatedOperatorType, OperatorType, Rule, matchesRule } from './rules';
+
+describe('rules', () => {
+  describe('Operators', () => {
+    it('ObfuscatedOperatorTypes should match hashed OperatorTypes', () => {
+      expect(ObfuscatedOperatorType.GTE).toBe(getMD5Hash(OperatorType.GTE));
```

Comment:
Same comment here with hardcoded hashed values

Reply:
These aren't hardcoded on purpose: want to ensure in the future that these match after hashing. That is because it is not possible to create an enum type with computed string values (so `ObfuscatedOperatorType.GTE` is defined as the hardcoded hash, rather than as `getMD5Hash(OperatorType.GTE)`)

Reply:
Oh right--yes carry on here 😅